examples/impute/plot_iterative_imputer_variants_comparison.py

```diff
@@ -0,0 +1,129 @@
+"""
+=========================================================
+Imputing missing values with variants of IterativeImputer
+=========================================================
+
+The :class:`sklearn.impute.IterativeImputer` class is very flexible - it can be
+used with a variety of predictors to do round-robin regression, treating every
+variable as an output in turn.
+
+In this example we compare some predictors for the purpose of missing feature
+imputation with `IterativeImputer`::
+
+    RidgeCV: default
+    HuberRegressor: robust linear regression to reduce the impact of outliers
+    DecisionTreeRegressor: non-linear regression
+    RandomForestRegressor: equivalent to missForest in R
+    KNeighborsRegressor: comparable to other KNN imputation approaches
+
+Of particular interest is the ability of ``IterativeImputer`` to mimic the
+behavior of missForest, a popular imputation package for R.
+
+The goal is to compare different predictors to see which one is best for
+the `IterativeImputer` when using a ``RandomForestRegressor`` estimator on the
+Boston dataset.
+
+For the Boston dataset, the ``HuberRegressor`` produces results that are on
+average superior to even having the full dataset. We also see that using other
```

Comment:
I think "superior" is a very strong word, even if qualified by "on average". Yes, this statement makes the plot a bit easier to read, but there is no gain due to imputation that even approaches significant. But the dataset is small.

Can we identify a dataset (real or generated) other than Boston where we might actually see significant or meaningful differences across methods?

Reply:
I think real would be better. What are some large-enough regression datasets that you've seen that would be appropriate here? This could end up being a slow script - I don't want to add 5m to sklearn's build time.

Reply:
Also, after I switched the code to use your missing pattern with `rng.choice`, now the `RandomForestRegressor` is best. It's quite sensitive to the missingness pattern, so maybe it's best to just leave it as is.